src/gui/DatabaseTabWidget.cpp

```diff
@@ -579,21 +496,18 @@ void DatabaseTabWidget::updateTabName(Database* db)
 
     QString tabName;
 
-    if (dbStruct.saveToFilename || dbStruct.readOnly) {
+    if (dbStruct.fileInfo.exists()) {
         if (db->metadata()->name().isEmpty()) {
-            tabName = dbStruct.fileName;
```

Comment:
@droidmonkey was there a possibility that the database was `readOnly` and the `fileName` was not defined?

Reply:
I am not sure what you mean by this. ```readOnly``` is only defined (with these new fixes) when the database file cannot be opened using ```QIODevice::ReadWrite``` and it can be opened using ```QIODevice::ReadOnly``` (see [Line 152](https://github.com/keepassxreboot/keepassxc/pull/1231/files#diff-12868a26bb71c81630f009044d3929caR152)).

Reply:
@droidmonkey nevermind, I was thinking about a possible execution path that might have occurred with the previous code.